experimenter/experimenter/jetstream/models.py

```diff
@@ -201,6 +201,164 @@ def compute_significance(data_point: DataPoint) -> Significance:
+                        getattr(significance_to_branch, window)[
+                            window_index
+                        ] = significance
+
+                if window == AnalysisWindow.WEEKLY:
+                    data_point.set_window_index(window_index)
+
+                comparison_data = getattr(metric_data, branch_comparison)
+                if branch_comparison == BranchComparison.ABSOLUTE:
+                    if len(comparison_data.all) == 0:
+                        comparison_data.first = data_point
+
+                    # TODO: remove `if` when this bug is fixed:
+                    #    https://github.com/mozilla/experimenter/issues/10043
+                    # ignore duplicate results
+                    # - OVERALL can only have one
+                    # - WEEKLY should not have dupes for the same index
+                    if (
+                        window == AnalysisWindow.OVERALL and len(comparison_data.all) == 0
+                    ) or (
+                        window == AnalysisWindow.WEEKLY
+                        and len(
+                            [
+                                p
+                                for p in comparison_data.all
+                                if p.window_index == window_index
+                            ]
+                        )
+                        == 0
+                    ):
+                        comparison_data.all.append(data_point)
+
+                # this is effectively an `else`, but we'll check just in case
+                if comparison_to_branch is not None:
+                    pairwise_comparison_data = getattr(
+                        comparison_data, comparison_to_branch
+                    )
+                    if len(pairwise_comparison_data.all) == 0:
+                        pairwise_comparison_data.first = data_point
+
+                    # TODO: remove `if` when this bug is fixed:
+                    #    https://github.com/mozilla/experimenter/issues/10043
+                    # ignore duplicate results
+                    # - OVERALL can only have one
+                    # - WEEKLY should not have dupes for the same index
+                    if (
+                        window == AnalysisWindow.OVERALL
+                        and len(pairwise_comparison_data.all) == 0
+                    ) or (
+                        window == AnalysisWindow.WEEKLY
+                        and len(
+                            [
+                                p
+                                for p in pairwise_comparison_data.all
+                                if p.window_index == window_index
+                            ]
+                        )
+                        == 0
+                    ):
+                        pairwise_comparison_data.all.append(data_point)
+
+    def append_conversion_count(self, primary_metrics_set):
+        for branch_name in self.dict():
+            branch = getattr(self, branch_name)
```

Comment:
```suggestion
        for branch_name, branch in self.dict().items():
```

Is this the preferred way to iterate over fields in Pydantic? It seems expensive to construct a dict from the object, potentially recursively, just to get field names.

It looks like there might be a `__fields__` member that you can iterate over (self.__fields__.keys()) to get the field names

Reply:
I agree that it's less than ideal, or at least looks odd, but there is a reason for this construction -- we want `branch` to be a `Branch` object (and `branch_data` to be a `BranchData`), and calling `self.dict()` converts the entire model to dictionaries. Doing `getattr(self, branch_name)` for each of the keys leave us with the `Branch` model instead of a dict.

That said, we could get the keys more efficiently, as you mentioned. I'll update to iterate `self.__fields__` instead of `self.dict()` (there's also a `__fields_set__`, which only includes fields set at instantiation, but given the custom constructor and dynamic model, I think it's safer to use `__fields__` in this case).